packages/@aws-cdk/aws-secretsmanager/lib/secret-target-attachment.ts

```diff
@@ -0,0 +1,71 @@
+import cdk = require('@aws-cdk/cdk');
+import { ISecret, Secret } from './secret';
+import { CfnSecretTargetAttachment } from './secretsmanager.generated';
+
+/**
+ * The type of service or database that's being associated with the secret.
+ */
+export enum AttachmentTargetType {
+  /**
+   * A database instance
+   */
+  Instance = 'AWS::RDS::DBInstance',
+
+  /**
+   * A database cluster
+   */
+  Cluster = 'AWS::RDS::DBCluster'
+}
+
+/**
+ * Options to add a secret attachement to a secret.
+ */
+export interface SecretTargetAttachmentOptions {
```

Comment:
Feel like this should be an interface like:

```ts
export interface ISecretAttachmentTarget {
  asSecretAttachmentTarget(): SecretAttachmentTargetProps;
}

export interface SecretAttachmentTargetProps {
   targetId: string;
   targetType: AttachmentTargetType;
}

export interface SecretTargetAttachmentOptions {
  target: ISecretAttachmentTarget;
}
```

And then have the DBInstance and DBCluster classes implement `ISecretAttachmentTarget`.